examples/correct/cursed.ccp

```diff
@@ -0,0 +1,17 @@
+let x: Unit = while true do break;
+
+let y: Int = (((((0)))));
+
+while true do while (break; true) do 1 / 0;
```

Comment:
I'd add some comment indicating what happens here, I guess the `while (break; true)` exists the outer loop?